Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Specific/ImageAttachmentPreviewPass.cpp

```diff
@@ -371,7 +371,14 @@ namespace AZ
         {
             // add attachments to the scope
             // input attachment
-            frameGraph.UseAttachment(RHI::ImageScopeAttachmentDescriptor{ m_imageAttachmentId }, RHI::ScopeAttachmentAccess::Read, RHI::ScopeAttachmentUsage::Shader);
+            RHI::FrameGraphAttachmentInterface attachmentDatabase = frameGraph.GetAttachmentDatabase();
+            RHI::ImageDescriptor imageDesc = attachmentDatabase.GetImageDescriptor(m_imageAttachmentId);
+            auto scopeAttachmentDesc = RHI::ImageScopeAttachmentDescriptor{ m_imageAttachmentId };
+            // aspect flag needs to be set for SRV (espectially for image contains both depth and stencil)
+            // Note: only support preview depth here if the image contains both depth and stencil
+            scopeAttachmentDesc.m_imageViewDescriptor.m_aspectFlags = RHI::CheckBitsAll(RHI::GetImageAspectFlags(imageDesc.m_format), RHI::ImageAspectFlags::Depth)?
```

Comment:
1. So if the image is DepthStencil we want RHI::ImageAspectFlags::Color?

2. Can you add a space on the either side for colon  "....Depth:RHI....". It was confusing to read otherwise

Reply:
+1 on the depth stencil usage - connects to my question above

Reply:
1.  I think it won't work since the image is created with depthstencil usage. 
2.  Sure

Reply:
your comment is what is confusing to me since you are using CheckBitsAll. Shouldnt it say

 Note: only support preview depth here if the image contains **only** Depth

Reply:
Good find. It should be CheckBitsAny. I would do some tests